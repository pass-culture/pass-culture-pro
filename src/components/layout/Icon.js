import PropTypes from 'prop-types'
import React from 'react'

import { ROOT_PATH } from '../../utils/config'

export const Icon = ({ svg, ...imgProps }) => (
  <img alt={svg} src={`${ROOT_PATH}/icons/${svg}.svg`} {...imgProps} />
)

Icon.propTypes = {
  svg: PropTypes.string.isRequired,
}

export default Icon
