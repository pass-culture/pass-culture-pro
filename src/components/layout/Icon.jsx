import PropTypes from 'prop-types'
import React from 'react'

import { ROOT_PATH } from '../../utils/config'

export const getImageUrl = (svg, png) => {
  let iconUrl
  svg ? (iconUrl = `${ROOT_PATH}/icons/${svg}.svg`) : (iconUrl = `${ROOT_PATH}/icons/${png}.png`)
  return iconUrl
}

const Icon = ({ png, svg, ...imgProps }) => {
  let altMessage
  const iconUrl = getImageUrl(svg, png)
  svg ? (altMessage = svg) : (altMessage = png)
  return (<img
    alt={altMessage}
    src={iconUrl}
    {...imgProps}
          />)
}

Icon.defaultProps = {
  png: null,
}

Icon.propTypes = {
  png: PropTypes.string,
  svg: PropTypes.string.isRequired,
}

export default Icon
