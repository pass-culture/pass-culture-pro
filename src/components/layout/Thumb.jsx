import PropTypes from 'prop-types'
import React from 'react'

import { SVGOffers } from '../svg/SVGOffers'

const Thumb = ({ url, alt = '' }) => {
  return url ? (
    <img
      alt={alt}
      className="offer-thumb"
      src={url}
    />
  ) : (
    <div className="default-thumb">
      <SVGOffers alt={alt} />
    </div>
  )
}

Thumb.defaultProps = {
  url: '',
}

Thumb.propTypes = {
  url: PropTypes.string,
}

export default Thumb
