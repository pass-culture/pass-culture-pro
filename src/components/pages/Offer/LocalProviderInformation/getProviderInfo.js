import { PROVIDER_ICONS } from '../../../utils/providers'

export const getProviderInfo = providerName => {
  const providers = [
    {
      id: 'allociné',
      icon: PROVIDER_ICONS['AllocineStocks'],
      name: 'Allociné',
    },
    {
      id: 'leslibraires.fr',
      icon: PROVIDER_ICONS['LibrairesStocks'],
      name: 'Leslibraires.fr',
    },
    {
      id: 'titelive',
      icon: PROVIDER_ICONS['TiteLiveStocks'],
      name: 'Tite Live',
    },
    {
      id: 'fnac',
      icon: PROVIDER_ICONS['FnacStocks'],
      name: 'Fnac',
    },
    {
      id: 'praxiel',
      icon: PROVIDER_ICONS['PraxielStocks'],
      name: 'Praxiel',
    },
  ]

  return providers.find(providerInfo => providerName.startsWith(providerInfo.id))
}
