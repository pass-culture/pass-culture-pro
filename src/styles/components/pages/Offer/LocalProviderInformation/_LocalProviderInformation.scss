.provider-information {
  display: flex;
  margin-bottom: 1.5rem;
}

.provider-thumb {
  display: flex;
}

.provider-details {
  padding-left: 1rem;
  display: block;
}

.provider-header {
  display: flex;
  align-items: center;

  .provider-title {
    padding-left: 0.75rem;

    h3 {
      display: inline-block;
      vertical-align: middle;
      padding-bottom: rem(1px);
    }

    span {
      vertical-align: middle;
      padding-left: 0.5rem;
      padding-top: rem(2px);
    }
  }
}

.provider-explanation {
  @extend .small-text;

  text-align: justify;
}
