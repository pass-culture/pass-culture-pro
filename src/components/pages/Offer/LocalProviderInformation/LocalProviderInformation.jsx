import React, { PureComponent } from 'react'
import Thumb from '../../../layout/Thumb'
import Icon from '../../../layout/Icon'
import ReactTooltip from 'react-tooltip'
import PropTypes from 'prop-types'

class LocalProviderInformation extends PureComponent {
  componentDidMount() {
    ReactTooltip.rebuild()
  }

  render() {
    const { offerName, offererId, providerInfo, thumbUrl, venueId } = this.props

    const thumbAltDescription = 'couverture du livre ' + offerName
    const tooltip =
      '<div>' +
      '<p>Vous pouvez modifier vos choix de synchronisation sur' +
      " <a href='/structures/" +
      offererId +
      '/lieux/' +
      venueId +
      "'>la page du lieu</a>" +
      '  qui présente cette offre, dans la section "Importation d’offres".' +
      '</p>' +
      '</div>'

    return (
      <div className="provider-information">
        <div className="provider-thumb">
          <Thumb
            alt={thumbAltDescription}
            src={thumbUrl}
          />
        </div>
        <div className="provider-details">
          <div className="provider-header">
            <Icon
              height="64px"
              svg={providerInfo.icon}
              width="64px"
            />
            <div className="provider-title">
              <h3>
                {`Offre synchronisée avec ${providerInfo.name}`}
              </h3>
              <span
                className="button"
                data-place="bottom"
                data-tip={tooltip}
                data-type="info"
              >
                <Icon svg="picto-info" />
              </span>
            </div>
          </div>
          <div className="provider-explanation">
            <p>
              {'Le visuel par défaut, les informations et le stock de cette offre sont synchronisés' +
              ` avec les données ${providerInfo.name} tous les soirs.`}
            </p>
          </div>
        </div>
      </div>
    )
  }
}

LocalProviderInformation.defaultProps = {
  thumbUrl: null,
}

LocalProviderInformation.propTypes = {
  offerName: PropTypes.string.isRequired,
  offererId: PropTypes.string.isRequired,
  providerInfo: PropTypes.shape().isRequired,
  thumbUrl: PropTypes.string,
  venueId: PropTypes.string.isRequired,
}

export default LocalProviderInformation
