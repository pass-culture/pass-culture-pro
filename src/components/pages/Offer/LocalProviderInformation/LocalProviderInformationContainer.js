import { connect } from 'react-redux'
import { compose } from 'redux'

import LocalProviderInformation from './LocalProviderInformation'
import { getProviderInfo } from './getProviderInfo'
import withFrenchQueryRouter from '../../../hocs/withFrenchQueryRouter'
import { selectOfferById } from '../../../../selectors/data/offersSelectors'

export const mapStateToProps = (state, ownProps) => {
  const {
    match: {
      params: { offerId },
    },
    isAllocine,
    isLibraires,
    isTitelive,
    offererId,
  } = ownProps

  const offer = selectOfferById(state, offerId)
  const { name: offerName, venueId, thumbUrl } = offer
  const providerInfo = getProviderInfo(isTitelive, isAllocine, isLibraires)

  return {
    offererId,
    offerName,
    providerInfo,
    thumbUrl,
    venueId,
  }
}

export default compose(
  withFrenchQueryRouter,
  connect(mapStateToProps)
)(LocalProviderInformation)
