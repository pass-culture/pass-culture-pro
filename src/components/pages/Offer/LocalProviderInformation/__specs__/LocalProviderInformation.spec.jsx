import { mount } from 'enzyme'
import React from 'react'

import Icon from 'components/layout/Icon'
import Thumb from 'components/layout/Thumb'

import LocalProviderInformation from '../LocalProviderInformation'

describe('src | LocalProviderInformationContainer', () => {
  let props

  beforeEach(() => {
    props = {
      offererId: 'ABCD',
      offerName: 'Super Livre',
      providerInfo: {
        icon: 'localProviderStocks',
        name: 'localProvider',
      },
      thumbUrl: 'http://localhost/storage/thumbs/products/AERTR',
      venueId: 'EARZ',
    }
  })

  it('should display an offer image', () => {
    // when
    const wrapper = mount(<LocalProviderInformation {...props} />)

    // then
    const thumb = wrapper.find(Thumb)
    expect(thumb.prop('url')).toBe('http://localhost/storage/thumbs/products/AERTR')
  })

  it('should display a link to venue page', () => {
    // when
    const wrapper = mount(<LocalProviderInformation {...props} />)

    // then
    const tooltip = wrapper.find('span').at(1)
    expect(tooltip.prop('data-tip')).toContain('<a href="/structures/ABCD/lieux/EARZ">')
  })

  it('should display provider information', () => {
    // Given
    const wrapper = mount(<LocalProviderInformation {...props} />)

    // When
    const title = wrapper.find({ children: 'Offre synchronisée avec localProvider' })
    const paragraphe = wrapper.find({
      children:
        'Le visuel par défaut, les informations et le stock de cette offre sont synchronisés avec les données localProvider tous les soirs.',
    })
    const logo = wrapper.find(Icon)

    // Then
    expect(title).toHaveLength(1)
    expect(logo.at(0).prop('svg')).toBe('localProviderStocks')
    expect(paragraphe).toHaveLength(1)
  })
})
