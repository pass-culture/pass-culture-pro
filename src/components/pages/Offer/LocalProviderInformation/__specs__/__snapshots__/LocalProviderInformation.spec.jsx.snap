// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Offer | LocalProviderInformationContainer | LocalProviderInformationContainer should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LocalProviderInformation
    offerName="Super Livre"
    offererId="ABCD"
    providerInfo={
      Object {
        "icon": "TiteliveStocks",
        "name": "Tite live",
      }
    }
    thumbUrl="http://localhost/storage/thumbs/products/AERTR"
    venueId="EARZ"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="provider-thumb"
        >
          <Thumb
            alt="couverture du livre Super Livre"
            src="http://localhost/storage/thumbs/products/AERTR"
          />
        </div>,
        <div
          className="provider-details"
        >
          <div
            className="provider-header"
          >
            <Icon
              height="64px"
              png={null}
              svg="TiteliveStocks"
              width="64px"
            />
            <div
              className="provider-title"
            >
              <h3>
                Offre synchronisée avec Tite live
              </h3>
              <span
                className="button"
                data-place="bottom"
                data-tip="<div><p>Vous pouvez modifier vos choix de synchronisation sur <a href='/structures/ABCD/lieux/EARZ'>la page du lieu</a>  qui présente cette offre, dans la section \\"Importation d’offres\\".</p></div>"
                data-type="info"
              >
                <Icon
                  png={null}
                  svg="picto-info"
                />
              </span>
            </div>
          </div>
          <div
            className="provider-explanation"
          >
            <p>
              Le visuel par défaut, les informations et le stock de cette offre sont synchronisés avec les données Tite live tous les soirs.
            </p>
          </div>
        </div>,
      ],
      "className": "provider-information",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Thumb
            alt="couverture du livre Super Livre"
            src="http://localhost/storage/thumbs/products/AERTR"
          />,
          "className": "provider-thumb",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "couverture du livre Super Livre",
            "src": "http://localhost/storage/thumbs/products/AERTR",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="provider-header"
            >
              <Icon
                height="64px"
                png={null}
                svg="TiteliveStocks"
                width="64px"
              />
              <div
                className="provider-title"
              >
                <h3>
                  Offre synchronisée avec Tite live
                </h3>
                <span
                  className="button"
                  data-place="bottom"
                  data-tip="<div><p>Vous pouvez modifier vos choix de synchronisation sur <a href='/structures/ABCD/lieux/EARZ'>la page du lieu</a>  qui présente cette offre, dans la section \\"Importation d’offres\\".</p></div>"
                  data-type="info"
                >
                  <Icon
                    png={null}
                    svg="picto-info"
                  />
                </span>
              </div>
            </div>,
            <div
              className="provider-explanation"
            >
              <p>
                Le visuel par défaut, les informations et le stock de cette offre sont synchronisés avec les données Tite live tous les soirs.
              </p>
            </div>,
          ],
          "className": "provider-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Icon
                  height="64px"
                  png={null}
                  svg="TiteliveStocks"
                  width="64px"
                />,
                <div
                  className="provider-title"
                >
                  <h3>
                    Offre synchronisée avec Tite live
                  </h3>
                  <span
                    className="button"
                    data-place="bottom"
                    data-tip="<div><p>Vous pouvez modifier vos choix de synchronisation sur <a href='/structures/ABCD/lieux/EARZ'>la page du lieu</a>  qui présente cette offre, dans la section \\"Importation d’offres\\".</p></div>"
                    data-type="info"
                  >
                    <Icon
                      png={null}
                      svg="picto-info"
                    />
                  </span>
                </div>,
              ],
              "className": "provider-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "height": "64px",
                  "png": null,
                  "svg": "TiteliveStocks",
                  "width": "64px",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h3>
                      Offre synchronisée avec Tite live
                    </h3>,
                    <span
                      className="button"
                      data-place="bottom"
                      data-tip="<div><p>Vous pouvez modifier vos choix de synchronisation sur <a href='/structures/ABCD/lieux/EARZ'>la page du lieu</a>  qui présente cette offre, dans la section \\"Importation d’offres\\".</p></div>"
                      data-type="info"
                    >
                      <Icon
                        png={null}
                        svg="picto-info"
                      />
                    </span>,
                  ],
                  "className": "provider-title",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Offre synchronisée avec Tite live",
                    },
                    "ref": null,
                    "rendered": "Offre synchronisée avec Tite live",
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <Icon
                        png={null}
                        svg="picto-info"
                      />,
                      "className": "button",
                      "data-place": "bottom",
                      "data-tip": "<div><p>Vous pouvez modifier vos choix de synchronisation sur <a href='/structures/ABCD/lieux/EARZ'>la page du lieu</a>  qui présente cette offre, dans la section \\"Importation d’offres\\".</p></div>",
                      "data-type": "info",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "png": null,
                        "svg": "picto-info",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": "span",
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                Le visuel par défaut, les informations et le stock de cette offre sont synchronisés avec les données Tite live tous les soirs.
              </p>,
              "className": "provider-explanation",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Le visuel par défaut, les informations et le stock de cette offre sont synchronisés avec les données Tite live tous les soirs.",
              },
              "ref": null,
              "rendered": "Le visuel par défaut, les informations et le stock de cette offre sont synchronisés avec les données Tite live tous les soirs.",
              "type": "p",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="provider-thumb"
          >
            <Thumb
              alt="couverture du livre Super Livre"
              src="http://localhost/storage/thumbs/products/AERTR"
            />
          </div>,
          <div
            className="provider-details"
          >
            <div
              className="provider-header"
            >
              <Icon
                height="64px"
                png={null}
                svg="TiteliveStocks"
                width="64px"
              />
              <div
                className="provider-title"
              >
                <h3>
                  Offre synchronisée avec Tite live
                </h3>
                <span
                  className="button"
                  data-place="bottom"
                  data-tip="<div><p>Vous pouvez modifier vos choix de synchronisation sur <a href='/structures/ABCD/lieux/EARZ'>la page du lieu</a>  qui présente cette offre, dans la section \\"Importation d’offres\\".</p></div>"
                  data-type="info"
                >
                  <Icon
                    png={null}
                    svg="picto-info"
                  />
                </span>
              </div>
            </div>
            <div
              className="provider-explanation"
            >
              <p>
                Le visuel par défaut, les informations et le stock de cette offre sont synchronisés avec les données Tite live tous les soirs.
              </p>
            </div>
          </div>,
        ],
        "className": "provider-information",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Thumb
              alt="couverture du livre Super Livre"
              src="http://localhost/storage/thumbs/products/AERTR"
            />,
            "className": "provider-thumb",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "alt": "couverture du livre Super Livre",
              "src": "http://localhost/storage/thumbs/products/AERTR",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="provider-header"
              >
                <Icon
                  height="64px"
                  png={null}
                  svg="TiteliveStocks"
                  width="64px"
                />
                <div
                  className="provider-title"
                >
                  <h3>
                    Offre synchronisée avec Tite live
                  </h3>
                  <span
                    className="button"
                    data-place="bottom"
                    data-tip="<div><p>Vous pouvez modifier vos choix de synchronisation sur <a href='/structures/ABCD/lieux/EARZ'>la page du lieu</a>  qui présente cette offre, dans la section \\"Importation d’offres\\".</p></div>"
                    data-type="info"
                  >
                    <Icon
                      png={null}
                      svg="picto-info"
                    />
                  </span>
                </div>
              </div>,
              <div
                className="provider-explanation"
              >
                <p>
                  Le visuel par défaut, les informations et le stock de cette offre sont synchronisés avec les données Tite live tous les soirs.
                </p>
              </div>,
            ],
            "className": "provider-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Icon
                    height="64px"
                    png={null}
                    svg="TiteliveStocks"
                    width="64px"
                  />,
                  <div
                    className="provider-title"
                  >
                    <h3>
                      Offre synchronisée avec Tite live
                    </h3>
                    <span
                      className="button"
                      data-place="bottom"
                      data-tip="<div><p>Vous pouvez modifier vos choix de synchronisation sur <a href='/structures/ABCD/lieux/EARZ'>la page du lieu</a>  qui présente cette offre, dans la section \\"Importation d’offres\\".</p></div>"
                      data-type="info"
                    >
                      <Icon
                        png={null}
                        svg="picto-info"
                      />
                    </span>
                  </div>,
                ],
                "className": "provider-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "height": "64px",
                    "png": null,
                    "svg": "TiteliveStocks",
                    "width": "64px",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h3>
                        Offre synchronisée avec Tite live
                      </h3>,
                      <span
                        className="button"
                        data-place="bottom"
                        data-tip="<div><p>Vous pouvez modifier vos choix de synchronisation sur <a href='/structures/ABCD/lieux/EARZ'>la page du lieu</a>  qui présente cette offre, dans la section \\"Importation d’offres\\".</p></div>"
                        data-type="info"
                      >
                        <Icon
                          png={null}
                          svg="picto-info"
                        />
                      </span>,
                    ],
                    "className": "provider-title",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Offre synchronisée avec Tite live",
                      },
                      "ref": null,
                      "rendered": "Offre synchronisée avec Tite live",
                      "type": "h3",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <Icon
                          png={null}
                          svg="picto-info"
                        />,
                        "className": "button",
                        "data-place": "bottom",
                        "data-tip": "<div><p>Vous pouvez modifier vos choix de synchronisation sur <a href='/structures/ABCD/lieux/EARZ'>la page du lieu</a>  qui présente cette offre, dans la section \\"Importation d’offres\\".</p></div>",
                        "data-type": "info",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "png": null,
                          "svg": "picto-info",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": "span",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  Le visuel par défaut, les informations et le stock de cette offre sont synchronisés avec les données Tite live tous les soirs.
                </p>,
                "className": "provider-explanation",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Le visuel par défaut, les informations et le stock de cette offre sont synchronisés avec les données Tite live tous les soirs.",
                },
                "ref": null,
                "rendered": "Le visuel par défaut, les informations et le stock de cette offre sont synchronisés avec les données Tite live tous les soirs.",
                "type": "p",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
